src/main/java/ch/njol/skript/conditions/CondFireworkHasDetonated.java

```diff
@@ -0,0 +1,70 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.Firework;
+import org.bukkit.entity.Projectile;
+
+@Name("Firework Is Detonated")
+@Description("Checks whether a firework has been detonated.")
+@Examples({
+	"while {_firework} has not been detonated:",
+		"\tspawn tnt at {_firework}",
+		"\twait 10 ticks"
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("MC 1.19+")
+public class CondFireworkHasDetonated extends PropertyCondition<Projectile> {
+
+	static {
+		if (Skript.methodExists(Firework.class, "isDetonated"))
+			Skript.registerCondition(CondFireworkHasDetonated.class, "%projectiles% (had|has|have) [:not] [been] detonated");
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setExpr((Expression<? extends Projectile>) exprs[0]);
+		setNegated(parseResult.hasTag("not"));
+		return true;
```

Comment:
```suggestion
		setNegated(parseResult.hasTag("not"));
		return super.init(exprs, matchedPattern, isDelayed, parseResult);
```

Reply:
I can just remove this part of the code as of applying the above requested changes which uses `PropertyCondition#register()`

Reply:
Disregard above comment, the negation pattern is not gramatically correct when registering with `PropertyType.HAVE`
![image](https://github.com/SkriptLang/Skript/assets/72163224/bad68b80-04f5-4a00-af35-452818ddf480)